### PR TITLE
Added new DO fields and made "name" a required parameter for machine.

### PR DESCRIPTION
**Pull request by will-chan**
Also made vpcId optional given that it's either vpcId or subnetId that
is required.

### CODE REVIEW

**Comment by cjellick** [comment]
Assuming tests pass, LGTM

File: code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json [hunk 1]
```diff
@@ -12,8 +12,7 @@
         },
         "vpcId": {
             "type": "string",
```
Comment (alena1108) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: We have to add an API filter to check if either vpcId or subnetId is specified, but not both

Comment (vincent99) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: Setting both seems to be valid in the driver (if not terribly useful, since subnet implies vpc).  There should be validation on our side that one or the other is set though if possible instead of going off and failing in machine.

Comment (cjellick) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: We've been taking the approach of only requiring the fields minimally required by docker-machine, Everything else, we allow machine to validate and then we propagate the error back up to the user.

Adding what you're suggeting isnt a bad idea though...especially as the drivers mature and stabalize.

Would you be comfortable with doing that in a separate PR further down the road?

Comment (vincent99) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: That's fine, I can add a check for that in the UI too.

Comment (alena1108) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: LGTM if we decide to move the validation part to the UI, or decide that we don't need it at all. @will-chan ?

Comment (will-chan) [inline code/implementation/docker/machine/src/main/resources/schema/base/amazonec2Config.json.d/amazonec2.json]: yeah, I think we should add the check in the UI but for now, since it's so specific to amazon, we'll let docker-machine handle it.  I don't want to get a place where docker-machine changes it's requirement on us and we have to change cattle just to accommodate it.

This does mean we might need to think about how far we want to consider supporting these "first class" drivers.  There's so much more we can do here before sending the values down to machine.

**Comment by alena1108** [comment]
LGTM

**Comment by cjellick** [comment]
Green build and 2 LGTMs. merging
